src/main/default/aura/Lookup/Lookup.cmp

```diff
@@ -1,79 +1,104 @@
-        <aura:attribute name="serverAction" type="Aura.Action" required="true" description="Server-side action for the search. Action must return an Apex List&lt;LookupSearchResult&gt;."/>
+        <aura:attribute name="serverAction" type="Aura.Action" required="true" description="Server-side action for the search. Action must return an Apex List&lt;LookupSearchResult&gt;." />
     </aura:method>
 
     <div class="slds-form-element">
         <label class="slds-form-element__label" for="{# globalId + '_combobox' }">{!v.label}</label>
         <div class="slds-form-element__control">
-            <div class="{! 'slds-combobox_container slds-has-inline-listbox '+ (v.hasFocus ? 'slds-has-input-focus' : '') }">
+            <div class="{! 'slds-combobox_container slds-has-inline-listbox ' + (and(v.hasFocus, empty(v.errors)) ? 'slds-has-input-focus' : '') + (!empty(v.errors) ? ' has-custom-error' : '') }">
                 <div class="{! 'slds-combobox slds-dropdown-trigger slds-dropdown-trigger_click '+ (and(v.hasFocus, !empty(v.searchResults)) ? 'slds-is-open' : 'slds-combobox-lookup') }"
                     aria-expanded="{! !empty(v.searchResults) }" aria-haspopup="listbox" role="combobox">
 
                     <aura:if isTrue="{!v.isMultiEntry}">
 
                         <!-- Multi-entry combobox start -->
                         <div class="slds-combobox__form-element slds-input-has-icon slds-input-has-icon_right" role="none">
-                            <input type="text" class="slds-input slds-combobox__input" aria-autocomplete="list" aria-controls="{# globalId + '_listbox' }" autocomplete="off" role="textbox"
-                                id="{# globalId + '_combobox' }"
-                                aura:id="searchInput"
-                                placeholder="{!v.placeholder}"
-                                value=""
-                                onfocus="{!c.onFocus}"
-                                onblur="{!c.onBlur}"
-                                oninput="{!c.onInput}"/>
-                            <lightning:icon iconName="utility:search" size="x-small" alternativeText="Search icon"
-                                class="slds-input__icon slds-input__icon_right"/>
+                            <input type="text" class="{! 'slds-input slds-combobox__input has-custom-height ' + (!empty(v.errors) ? 'has-custom-error' : '') }"
+                                aria-autocomplete="list" aria-controls="{# globalId + '_listbox' }" autocomplete="off"
+                                role="textbox" id="{# globalId + '_combobox' }" aura:id="searchInput" placeholder="{!v.placeholder}"
+                                value="" onfocus="{!c.onFocus}" onblur="{!c.onBlur}" oninput="{!c.onInput}" />
+
+                            <!-- search icon and spinner -->
+                            <aura:if isTrue="{!v.isLoading}">
```

Comment:
Please use conditional style with `slds-hide` class instead of `aura:if` in this context.

`aura:if` destroys the underlying DOM when toggled off so this is not ideal for something that frequently changes like the loading icon. Better use CSS instead.